### PR TITLE
Enable Form inputs to manipulate related field values

### PR DESCRIPTION
**Pull request by luacmartins**
### Details
- Allows Form a component to directly manipulate the value of other related fields, e.g. AddressSearch, State, City, etc

### Fixed Issues
$ https://github.com/Expensify/App/issues/9059

### Tests
1. Change the [this path](https://github.com/Expensify/App/blob/671dd98de01e3046765c6262c8b726028b9cc07a/.storybook/webpack.config.js#L10) from `../.env.staging` to `../.env`, to avoid CORS errors with the `AddressSearch` component
2. Run `npm run storybook` and in the `Form` tests, enter an address and verify that the `State` field is updated as well.
3. In addition, test the Address search component in the following pages, making sure that error messages show up and that the value can be edited and the related inputs (City, State, Zip Code) are updated:
    - `Settings > Payments > Add payment method > Debit card`
    - `+ > Send money > Enter Amount > Select User > Pay with Expensify > Bank Account > Additional Details`
    - `Settings > Workspace > Connect bank account > Company information`
    - `Settings > Workspace > Connect bank account > Personal information`
    - `Settings > Workspace > Connect bank account > Additional information`

- [x] Verify that no errors appear in the JS console

#### Contributor (PR Author) Checklist
- [x] I linked the correct issue in the `### Fixed Issues` section above
- [x] I wrote clear testing steps that cover the changes made in this PR
    - [x] I added steps for local testing in the `Tests` section
    - [x] I added steps for Staging and/or Production testing in the `QA steps` section
    - [x] I added steps to cover failure scenarios (i.e. verify an input displays the correct error message if the entered data is not correct)
    - [ ] I turned off my network connection and tested it while offline to ensure it matches the expected behavior (i.e. verify the default avatar icon is displayed if app is offline)
- [x] I included screenshots or videos for tests on [all platforms](https://github.com/Expensify/App/blob/main/CONTRIBUTING.md#make-sure-you-can-test-on-all-platforms)
- [x] I ran the tests on **all platforms** & verified they passed on:
    - [x] iOS / native
    - [x] Android / native
    - [x] iOS / Safari
    - [x] Android / Chrome
    - [x] MacOS / Chrome
    - [x] MacOS / Desktop
- [x] I verified there are no console errors (if there’s a console error not related to the PR, report it or open an issue for it to be fixed)
- [x] I followed proper code patterns (see [Reviewing the code](https://github.com/Expensify/App/blob/main/PR_REVIEW_GUIDELINES.md#reviewing-the-code))
    - [ ] I verified that any callback methods that were added or modified are named for what the method does and never what callback they handle (i.e. `toggleReport` and not `onIconClick`)
    - [ ] I verified that comments were added to code that is not self explanatory
    - [ ] I verified that any new or modified comments were clear, correct English, and explained “why” the code was doing something instead of only explaining “what” the code was doing.
    - [ ] I verified any copy / text shown in the product was added in all `src/languages/*` files
    - [ ] I verified any copy / text that was added to the app is correct English and approved by marketing by tagging the marketing team on the original GH to get the correct copy.
    - [ ] I verified proper file naming conventions were followed for any new files or renamed files. All non-platform specific files are named after what they export and are not named “index.js”. All platform-specific files are named for the platform the code supports as outlined in the README.
    - [ ] I verified the JSDocs style guidelines (in [`STYLE.md`](https://github.com/Expensify/App/blob/main/STYLE.md#jsdocs)) were followed
- [ ] If a new code pattern is added I verified it was agreed to be used by multiple Expensify engineers
- [x] I followed the guidelines as stated in the [Review Guidelines](https://github.com/Expensify/App/blob/main/PR_REVIEW_GUIDELINES.md)
- [x] I tested other components that can be impacted by my changes (i.e. if the PR modifies a shared library or component like `Avatar`, I verified the components using `Avatar` are working as expected)
- [x] I verified all code is DRY (the PR doesn't include any logic written more than once, with the exception of tests)
- [x] I verified any variables that can be defined as constants (ie. in CONST.js or at the top of the file that uses the constant) are defined as such
- [ ] If a new component is created I verified that:
    - [ ] A similar component doesn't exist in the codebase
    - [ ] All props are defined accurately and each prop has a `/** comment above it */`
    - [ ] Any functional components have the `displayName` property
    - [ ] The file is named correctly
    - [ ] The component has a clear name that is non-ambiguous and the purpose of the component can be inferred from the name alone
    - [ ] The only data being stored in the state is data necessary for rendering and nothing else
    - [ ] For Class Components, any internal methods passed to components event handlers are bound to `this` properly so there are no scoping issues (i.e. for `onClick={this.submit}` the method `this.submit` should be bound to `this` in the constructor)
    - [ ] Any internal methods bound to `this` are necessary to be bound (i.e. avoid `this.submit = this.submit.bind(this);` if `this.submit` is never passed to a component event handler like `onClick`)
    - [ ] All JSX used for rendering exists in the render method
    - [ ] The component has the minimum amount of code necessary for its purpose and it is
- [ ] If a new CSS style is added I verified that:
    - [ ] A similar style doesn’t already exist
    - [ ] The style can’t be created with an existing [StyleUtils](https://github.com/Expensify/App/blob/main/src/styles/StyleUtils.js) function (i.e. `StyleUtils.getBackgroundAndBorderStyle(themeColors.componentBG`)
- [x] If the PR modifies a generic component, I tested and verified that those changes do not break usages of that component in the rest of the App (i.e. if a shared library or component like `Avatar` is modified, I verified that `Avatar` is working as expected in all cases)
- [x] If the PR modifies a component related to any of the existing Storybook stories, I tested and verified all stories for that component are still working as expected.

<details>
<summary><h4>PR Reviewer Checklist</h4></summary>

- [x] I verified the correct issue is linked in the `### Fixed Issues` section above
- [x] I verified testing steps are clear and they cover the changes made in this PR
    - [x] I verified the steps for local testing are in the `Tests` section
    - [x] I verified the steps for Staging and/or Production testing are in the `QA steps` section
    - [x] I verified the steps cover any possible failure scenarios (i.e. verify an input displays the correct error message if the entered data is not correct)
    - [ ] I turned off my network connection and tested it while offline to ensure it matches the expected behavior (i.e. verify the default avatar icon is displayed if app is offline)
- [x] I checked that screenshots or videos are included for tests on [all platforms](https://github.com/Expensify/App/blob/main/CONTRIBUTING.md#make-sure-you-can-test-on-all-platforms)
- [x] I verified tests pass on **all platforms** & I tested again on:
    - [x] iOS / native
    - [ ] Android / native
    - [x] iOS / Safari
    - [ ] Android / Chrome
    - [x] MacOS / Chrome
    - [ ] MacOS / Desktop
- [x] I verified there are no console errors (if there’s a console error not related to the PR, report it or open an issue for it to be fixed)
- [x] I verified proper code patterns were followed (see [Reviewing the code](https://github.com/Expensify/App/blob/main/PR_REVIEW_GUIDELINES.md#reviewing-the-code))
    - [ ] I verified that any callback methods that were added or modified are named for what the method does and never what callback they handle (i.e. `toggleReport` and not `onIconClick`).
    - [ ] I verified that comments were added to code that is not self explanatory
    - [ ] I verified that any new or modified comments were clear, correct English, and explained “why” the code was doing something instead of only explaining “what” the code was doing.
    - [ ] I verified any copy / text shown in the product was added in all `src/languages/*` files
    - [ ] I verified any copy / text that was added to the app is correct English and approved by marketing by tagging the marketing team on the original GH to get the correct copy.
    - [ ] I verified proper file naming conventions were followed for any new files or renamed files. All non-platform specific files are named after what they export and are not named “index.js”. All platform-specific files are named for the platform the code supports as outlined in the README.
    - [ ] I verified the JSDocs style guidelines (in [`STYLE.md`](https://github.com/Expensify/App/blob/main/STYLE.md#jsdocs)) were followed
- [x] If a new code pattern is added I verified it was agreed to be used by multiple Expensify engineers
- [x] I verified that this PR follows the guidelines as stated in the [Review Guidelines](https://github.com/Expensify/App/blob/main/PR_REVIEW_GUIDELINES.md)
- [x] I verified all code is DRY (the PR doesn't include any logic written more than once, with the exception of tests)
- [x] I verified any variables that can be defined as constants (ie. in CONST.js or at the top of the file that uses the constant) are defined as such
- [x] If a new component is created I verified that:
    - [ ] A similar component doesn't exist in the codebase
    - [ ] All props are defined accurately and each prop has a `/** comment above it */`
    - [ ] Any functional components have the `displayName` property
    - [ ] The file is named correctly
    - [ ] The component has a clear name that is non-ambiguous and the purpose of the component can be inferred from the name alone
    - [ ] The only data being stored in the state is data necessary for rendering and nothing else
    - [ ] For Class Components, any internal methods passed to components event handlers are bound to `this` properly so there are no scoping issues (i.e. for `onClick={this.submit}` the method `this.submit` should be bound to `this` in the constructor)
    - [ ] Any internal methods bound to `this` are necessary to be bound (i.e. avoid `this.submit = this.submit.bind(this);` if `this.submit` is never passed to a component event handler like `onClick`)
    - [ ] All JSX used for rendering exists in the render method
    - [ ] The component has the minimum amount of code necessary for its purpose and it is broken down into smaller components in order to separate concerns and functions
- [ ] If a new CSS style is added I verified that:
    - [ ] A similar style doesn’t already exist
    - [ ] The style can’t be created with an existing [StyleUtils](https://github.com/Expensify/App/blob/main/src/styles/StyleUtils.js) function (i.e. `StyleUtils.getBackgroundAndBorderStyle(themeColors.componentBG`)
- [x] If the PR modifies a generic component, I tested and verified that those changes do not break usages of that component in the rest of the App (i.e. if a shared library or component like `Avatar` is modified, I verified that `Avatar` is working as expected in all cases)
- [x] If the PR modifies a component related to any of the existing Storybook stories, I tested and verified all stories for that component are still working as expected.

</details>

### QA Steps
1. Test the Address search functionality in the following pages, making sure that error messages show up and that the value can be edited and the related inputs (City, State, Zip Code) are updated:
    - `Settings > Payments > Add payment method > Debit card`
    - `+ > Send money > Enter Amount > Select User > Pay with Expensify > Bank Account > Additional Details`
    - `Settings > Workspace > Connect bank account > Company information`
    - `Settings > Workspace > Connect bank account > Personal information`
    - `Settings > Workspace > Connect bank account > Additional information`

- [ ] Verify that no errors appear in the JS console

#### Web

https://user-images.githubusercontent.com/22219519/171066623-a9f60f20-ce49-40de-be82-1f4c334fe6ff.mov

#### Mobile Web

https://user-images.githubusercontent.com/22219519/171066630-717e1c05-b965-4240-a59f-cc6f977f0956.mov

#### Desktop

https://user-images.githubusercontent.com/22219519/171066641-8131dd6e-8933-4487-b6e1-bcaa7c9acd4d.mov

#### iOS

https://user-images.githubusercontent.com/22219519/171066657-2acea9b6-2a72-469e-bd52-2a45241693d3.mov

#### Android

https://user-images.githubusercontent.com/22219519/171067112-730639d2-19fc-4deb-8546-09dab4735557.mov

#### Storybook

https://user-images.githubusercontent.com/22219519/171067118-92f0bb9c-a488-4f5d-bd27-edb226ea4885.mov

### CODE REVIEW

File: src/components/AddressSearch.js [hunk 1]
```diff
@@ -161,7 +165,7 @@ const AddressSearch = (props) => {
                         label: props.label,
                         containerStyles: props.containerStyles,
                         errorText: props.errorText,
-                        hint: props.hint,
+                        hint: displayListViewBorder ? undefined : props.hint,
```
Comment (luacmartins) [inline src/components/AddressSearch.js:168]: I added this because I noticed that the dropdown was showing below the hint, which looked odd.

**Review by Julesssss — APPROVED** [review]
Working pretty well. Obviously, this doesn't fill the state field when entering a UK address (as expected). But the ZIP code input field was updated.

**Review by Julesssss — APPROVED** [review]
Okay, I managed to test successfully.

**Comment by kavimuru** [comment]
@luacmartins We are not able to test  _Send money > Enter Amount > Select User > Pay with Expensify > Bank Account > Additional Details . We are blocked by this issue #7185

**Comment by luacmartins** [comment]
@kavimuru thanks for letting me know. I think that's ok if all the other flows were tested and passed. We should be good here.

**Comment by eVoloshchak** [comment]
⁦👋🏻 Coming here from https://github.com/Expensify/App/issues/16448, which was introduced by this PR
On this line, we're hiding the tooltip if `displayListViewBorder` is true
https://github.com/Expensify/App/blob/671dd98de01e3046765c6262c8b726028b9cc07a/src/components/AddressSearch.js#L168

If you select an item from the list, we manually set `displayListViewBorder` to false and the tooltip appears. But if you just press outside of the list, it closes, but `displayListViewBorder` is still true, the tooltip is hidden, which isn't correct. 

In short, this just surfaced a limitation of the initial implementation in https://github.com/Expensify/App/pull/6115. This wasn't needed in the original PR since there was no tooltip logic, but now it should also set `displayListViewBorder` to false in `onBlur` listener
